astropy/coordinates/tests/test_representation.py

```diff
@@ -283,6 +307,25 @@ def test_raise_on_extra_arguments(self):
         with pytest.raises(TypeError, match='unexpected keyword.*parrot'):
             SphericalRepresentation(1*u.deg, 2*u.deg, 1.*u.kpc, parrot=10)
 
+    def test_representation_shortcuts(self):
+        """Test that shortcuts in ``represent_as`` don't fail."""
+        difs = SphericalCosLatDifferential(4*u.mas/u.yr,5*u.mas/u.yr,6*u.km/u.s)
+        sph = SphericalRepresentation(1*u.deg, 2*u.deg, 3*u.kpc,
+                                      differentials={'s': difs})
+
+        got = sph.represent_as(PhysicsSphericalRepresentation,
+                               PhysicsSphericalDifferential)
+        expected = BaseRepresentation.represent_as(
```

Comment:
It may make sense here to test that the short-cut route was actually taken. Should be possible to do that with
```
assert np.may_share_memory(sph.distance, got.r)
assert np.may_share_memory(difs.d_distance, got.differentials['s'].d_r)

Reply:
That seems aa good way to test it. I was trying to find a way, other than just testing it didn't fail and inspecting the coverage xml.

Reply:
@mhvk. Resolved in force-push.